security-features/cryptography/legacy-tls.rst

```diff
@@ -1,8 +1,136 @@
+Ubuntu 20.04 LTS implemented a significant security enhancement by defaulting to **TLS v1.2 or higher** across all major TLS libraries. This change aligns with industry standards that phase out the insecure TLS 1.0 and 1.1 protocols, which contain known cryptographic vulnerabilities such as susceptibility to POODLE, BEAST, and downgrade attacks.
 
-For this reason, Ubuntu 20.04 and later proactively disable these protocols, requiring more secure alternatives.
+----
+
+Affected Libraries and Configuration Changes
+--------------------------------------------
+
+The following TLS libraries have been updated with new security defaults:
+
+**OpenSSL**
+    - **Security Level**: Raised from ``SECLEVEL=1`` to ``SECLEVEL=2``
+    - **Minimum TLS Version**: 1.2 (TLS 1.0/1.1 disabled)
+    - **Disabled Algorithms**: SHA-1 signatures, RSA keys <2048 bits, DH keys <2048 bits
+    - **Default Cipher String**: ``DEFAULT@SECLEVEL=2``
+
+**GnuTLS**
+    - **Priority String**: ``NORMAL:-VERS-ALL:+VERS-TLS1.3:+VERS-TLS1.2:+VERS-DTLS1.2:%PROFILE_MEDIUM``
+    - **Disabled Protocols**: TLS 1.0 and 1.1 explicitly excluded
+    - **Configuration File**: ``/etc/gnutls/config`` (if present)
+
+**NSS (Network Security Services)**
+    - **Minimum Protocol**: TLS 1.2 enforced internally
+    - **Weak Ciphers**: Legacy cipher suites disabled by default
+
+**Qt Framework**
+    - **Effective Impact**: Cannot override OpenSSL security levels programmatically
+    - **Behavior**: TLS 1.0/1.1 requests fail with socket error -1
+    - **API Limitation**: Setting lower TLS versions breaks connectivity entirely
+
+----
+
+Legacy TLS Re-enablement (Not Recommended)
+-------------------------------------------
+
+.. warning::
+   Re-enabling TLS 1.0/1.1 significantly reduces security and exposes systems to serious vulnerabilities. Only use as a temporary measure for critical legacy compatibility. Use at your own risk of data compromise.
+
+**OpenSSL System-wide Configuration**
+
+1. Edit ``/etc/ssl/openssl.cnf`` and add at the very top (before any sections)::
+
+    openssl_conf = default_conf
+
+2. Add the following sections at the end of the file::
+
+    [default_conf]
+    ssl_conf = ssl_sect
+
+    [ssl_sect]
+    system_default = system_default_sect
+
+    [system_default_sect]
+    CipherString = DEFAULT@SECLEVEL=1
+
+**OpenSSL Per-Application Override**
+
+Use environment variable to specify custom configuration::
+
+    export OPENSSL_CONF=/path/to/custom-openssl.cnf
+
+**GnuTLS Override**
+
+1. Create ``/etc/gnutls/config``::
+
+    [overrides]
+    default-priority-string = NORMAL
+
+2. Alternatively, use environment variable::
+
+    export GNUTLS_SYSTEM_PRIORITY_FILE=/path/to/gnutls-override-config
+
+----
+
+Common Issues and Troubleshooting
+----------------------------------
+
+**Connection Symptoms and Solutions**
```

Comment:
I think this and **Diagnostic Commands** would be better suited as sub-sections so they can be navigated to from the side bar.